### PR TITLE
feature: add helm chart

### PR DESCRIPTION
**Pull request by paulopontesm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Codacy Chart
+
+This is the top level chart to run Codacy in enterprise mode.
+With this chart, it is  possible to run all the components from Codacy,
```
Comment (h314to) [inline README.md]: extra space: `is  possible`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+3. Wide support of cloud service providers
+4. Use external data stores, such as Minio (S3) or Amazon RDS (Postgres), whenever possible
+5. Users should be able to use their own certificates or use Let's Encrypt
+6. Users should be able to use a load balancer or an ingess
+
+We will try to leverage standard Kubernetes features:
+
+* ConfigMaps for managing configuration. These will then get mapped or passed to
+Docker containers
+* Secrets for sensitive data
```
Comment (h314to) [inline README.md]: I would remove this. I think it is more related to planning than the actual finished chart.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Helm Charts
+
+![Helm Chart Structure](./images/charts.png)
+
+[edit image](https://docs.google.com/drawings/d/1o7z3L8XnnNjHBOTWKHiIYUkBP3DDiogdUyxNdUfzyfY/edit)
+
+Each service/component in Codacy has it's own chart published to `charts.codacy.com`.
+
+This particular chart will work as a way of bundling all the components
```
Comment (h314to) [inline README.md]: change tense: `This chart bundles all the components and their dependencies.`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+together with its dependencies. For the bundle we will make use of the
```
Comment (h314to) [inline README.md]: will make use -> make use

File: README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+[requirements capability](https://helm.sh/docs/chart_best_practices/#requirements-files)
+of Helm.
+
+Helm will help us manage this complexity, and provide an easy top level interface
+to manage settings via the `values.yaml` file.
+
```
Comment (h314to) [inline README.md]: Remove

File: .doks/Makefile [hunk 6]
```diff
@@ -0,0 +1,21 @@
+.PHONY: deploy_to_doks
+deploy_to_doks:
+	doctl kubernetes cluster kubeconfig save $${DOKS_CLUSTER_NAME} --set-current-context
+
+	helm repo add codacy-stable https://charts.codacy.com/stable
+	helm repo add codacy-unstable https://charts.codacy.com/unstable
+	helm repo add codacy-incubator https://charts.codacy.com/incubator
```
Comment (DReigada) [inline .doks/Makefile]: Does it make sense to have these two repos here?

Comment (DReigada) [inline .doks/Makefile]: Does it make sense to have these two repos here?

File: .doks/values.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+listener:
+  replicaCount: 4
+  listenerdb:
+    persistence:
+      enabled: true
+      size: 250Gi
+
+  persistence:
+    enabled: true
```
Comment (DReigada) [inline .doks/values.yaml]: This value is not used by the component

File: README.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
+# Codacy Chart
+
+This is the top level chart to run Codacy in enterprise mode.
```
Comment (DReigada) [inline README.md]: enterprise -> self-hosted

File: README.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+With this chart, it is possible to run all the components from Codacy,
+plugs all the dependencies with a single command line, including Codacy,
```
Comment (DReigada) [inline README.md]: plugs -> plugging

File: README.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+5.  Users should be able to use their own certificates or use Let's Encrypt
+6.  Users should be able to use a load balancer or an ingess
+
+We will try to leverage standard Kubernetes features:
+
+*   ConfigMaps for managing configuration. These will then get mapped or passed to Docker containers
+
+## Helm Charts
+
+![Helm Chart Structure](./images/charts.png)
```
Comment (DReigada) [inline README.md]: Use the following link https://docs.google.com/drawings/d/1o7z3L8XnnNjHBOTWKHiIYUkBP3DDiogdUyxNdUfzyfY/export/png

And make the doc public

File: codacy/Chart.yaml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+appVersion: "1.0"
+description: |
+  This is the top level chart to run Codacy in enterprise mode.
+  With this chart, it is possible to run all the components from Codacy,
+  plugs all the dependencies with a single command line, including Codacy,
```
Comment (DReigada) [inline codacy/Chart.yaml]: plugs -> plugging

File: codacy/values.yaml [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+    postgresqlUsername: codacy
+    postgresqlDatabase: hotspots
+    postgresqlPassword: codacy
+    service:
+      port: 5432
+
+hotspots-worker:
+  replicaCount: 1
+
+listener:
```
Comment (DReigada) [inline codacy/values.yaml]: The nfs provisioner settings are missing

File: .doks/Makefile (hunk 6 above)
Comment (paulopontesm) [reply .doks/Makefile]: yes, because in our .doks cluster we are installing stuff from incubator sometimes

File: README.md (hunk 10 above)
Comment (paulopontesm) [reply README.md]: I can't make things in our shared drive directory public. The option would be to move it to my drive but I could end up deleting it or something. I think ill just leave it like it is.

**Pull request merged**